src/MediaCollections/Filesystem.php

```diff
@@ -229,6 +229,16 @@ public function syncFileNames(Media $media): void
         $this->renameConversionFiles($media);
     }
 
+    public function syncMediaPath(Media $media): void
+    {
+        $factory = PathGeneratorFactory::create();
+
+        $oldMedia = (clone $media)->fill($media->getOriginal());
+
+        $this->filesystem->disk($media->disk)
```

Comment:
Maybe we should compare the old path and with the new path before calling `move`

Reply:
Yes, it's good. So, we can remove the config `moves_media_on_update` and call the sync method when the paths are not equals. Is it right?

Reply:
I would still only do the move if the `moves_media_on_update` is set to `true`. 

The base idea is that after upload, medialibrary won't any files, unless you explicitly enable it. Feel safer to me!